.github/workflows/package-manager-demo.yml

```diff
@@ -11,42 +11,51 @@ permissions:
 
 jobs:
   build:
-    env:
-      PACKAGEMANAGER_TOKEN_SOLO: ${{ secrets.PACKAGEMANAGER_TOKEN_SOLO }}
-      PACKAGEMANAGER_TOKEN_CLUSTER: ${{ secrets.PACKAGEMANAGER_TOKEN_CLUSTER }}
-      PACKAGEMANAGER_TOKEN_S3: ${{ secrets.PACKAGEMANAGER_TOKEN_S3 }}
     strategy:
       matrix:
         r-version: ['3.6.3', '4.1.3']
         os: [ {os: "ubuntu-20.04", distro: "focal"}, {os: "ubuntu-18.04", distro: "bionic"} ]
         package-manager: [
-          {token: "$PACKAGEMANAGER_TOKEN_SOLO", endpoint: "https://solo.rstudiopm.com"},
-          {token: "$PACKAGEMANAGER_TOKEN_S3", endpoint: "https://s3.rstudiopm.com"},
-          {token: "$PACKAGEMANAGER_TOKEN_CLUSTER", endpoint: "https://cluster.rstudiopm.com"},
+          {endpoint: "https://solo.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_SOLO"},
+          {endpoint: "https://s3.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_S3"},
+          {endpoint: "https://cluster.rstudiopm.com", token: "PACKAGEMANAGER_TOKEN_CLUSTER"},
         ]
     runs-on: ${{ matrix.os.os }}
     steps:
       - uses: actions/checkout@v3
       - name: Set up R ${{ matrix.r-version }}
-        uses: r-lib/actions/setup-r@f57f1301a053485946083d7a45022b278929a78a
+        uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.r-version }}
       - name: Build source package
-        run: R CMD build .
+        run: |
+          source_dir=$(mktemp -d)
+          (cd $source_dir; R CMD build $GITHUB_WORKSPACE)
+          echo "SOURCE_FILE=$(readlink -f $source_dir/*)" >> $GITHUB_ENV
         shell: bash
       - name: Build binary package
-        run: R CMD INSTALL --build .
+        run: |
+          binary_dir=$(mktemp -d)
+          (cd $binary_dir; R CMD INSTALL --build $GITHUB_WORKSPACE)
+          echo "BINARY_FILE=$(readlink -f $binary_dir/*)" >> $GITHUB_ENV
         shell: bash
       - name: Install RSPM CLI
-        run: "curl -O -J -H \"Authorization: Bearer ${{ matrix.package-manager.token }}\" ${{ matrix.package-manager.endpoint }}/__api__/download && chmod +x ./rspm"
```

Comment:
I think this was putting the secret in the command line. Unfortunately there doesn't seem to be a way to use secrets as environment variables in GH Actions, so I used this method, which also simplified the matrix a bit: https://github.community/t/matrix-based-secrets/166376/2